src/libs/masks.c

```diff
@@ -661,30 +661,58 @@ static void _tree_delete_shape(GtkButton *button, dt_lib_module_t *self)
 {
   dt_lib_masks_t *lm = (dt_lib_masks_t *)self->data;
 
+  dt_masks_clear_form_gui(darktable.develop);
+
   // now we go through all selected nodes
   GtkTreeModel *model = gtk_tree_view_get_model(GTK_TREE_VIEW(lm->treeview));
   GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(lm->treeview));
   dt_iop_module_t *module = NULL;
-  ++darktable.gui->reset;
+
   GList *items = gtk_tree_selection_get_selected_rows(selection, NULL);
-  for(const GList *items_iter = items; items_iter; items_iter = g_list_next(items_iter))
+
+  for(const GList *items_iter = items;
+      items_iter;
+      items_iter = g_list_next(items_iter))
   {
     GtkTreePath *item = (GtkTreePath *)items_iter->data;
     GtkTreeIter iter;
     if(gtk_tree_model_get_iter(model, &iter, item))
     {
+      GtkTreeIter *prev_iter = gtk_tree_iter_copy(&iter);
+      GtkTreeIter *next_iter = gtk_tree_iter_copy(&iter);
+      const gboolean has_previous = gtk_tree_model_iter_previous(model, prev_iter);
+      const gboolean has_next = gtk_tree_model_iter_next(model, next_iter);
+      int prev_grid = -1;
```

Comment:
Should be INVALID_MASKID here and below ?

Reply:
Indeed, and many other occurrences in this file. Now fixed. Thanks.